docs/DeveloperGuide.md

```diff
@@ -1,11 +1,82 @@
+
+## implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
+![Architect sequence diagram](UML diagrams/OverviewSequenceDiagram.png)
+
+### Login Component
+The login component consists of the classes that run the main functions of the Login user Interface
+![Login Component Class Diagram](UML diagrams/LoginComponentClassDiagram.png)
+The **Login Class Diagram** give above shows how the classes in the Login Component interact with each other
+
+The Login Component consists of 5 components.
+* <code> LoginController </code> The LoginController handles the login process and control logic of the login process.
+* <code> LoginManager </code> The LoginManager provides the logic to verify the user login details
+* <code> LoginUi </code> The LoginUi defines the display and gets input from the user. LoginUi extends the functionality of the Ui class
+* <code> LoginInfo </code> The LoginInfo contains the atrributes and methods of each LoginInfo object from a particular user logging into MojoHR application
+* <code> LoginInfoFileManager </code> The LoginInfoFileManager is responsible for the logic that handles the storage and retrieval of login information
+
+### Email Component
+The email component consists of classes that are involved with all the emails
+![Email Component Class Diagram](UML diagrams/EmailClassDiagram.png)
+The **Login Class Diagram** give above shows how the classes in the Email Component interact with each other
+
+The Email Component consists of 8 components.
+* <code> Email </code> is the parent class to the other email classes and contains all the information that is required in an email
+* <code> Archive </code> inherits from Email
+* <code> Deleted </code> inherits from Email
+* <code> Draft </code> inherits from Email
+* <code> Inbox </code> inherits from Email
+* <code> Junk </code> inherits from Email
+* <code> Sent </code> inherits from Email
+* <code> EmailManager </code> is responsible for the logic of retrieving Emails as a whole
+
+## Implementation
+
+### ListCommand 
+There are seven variations of the <code> list </code> command.
+1. List by All email(s)
+2. List by Archived email(s) 
+3. List by Deleted email(s)
+4. List by Draft(s) email(s) 
+5. List by Inbox email(s)
+6. List by Junk email(s)
+7. List by Sent email(s)
+
+The sequence diagram shows how the <code> list(type) </code> operation work. 
+![ListCommand Sequence Diagram](Sequence Diagrams/ListCommand.png)
+
+###ComposeCommand
+When the `compose` command is entered, the software will prompt user to enter the necessary details and save them to draft.
+
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
 
+The sequence diagram roughly shows how the `compose` operation works.
+![ComposeCommand Sequence Diagram](Sequence Diagrams/ComposeCommand.png)
```

Comment:
![image](https://user-images.githubusercontent.com/56064050/113082529-3cf74400-920d-11eb-9e9d-98eb37b1c208.png)

Is the return arrow for "7" necessary since the rest of the methods do not include return arrows?